CMakeLists.txt

```diff
@@ -0,0 +1,630 @@
+if(NOT LIBXMP_BUILD_LITE AND NOT LIBXMP_BUILD_FULL)
+    message(FATAL_ERROR "Both Full and Lite versions got disabled. You must enable at least one of them.")
+endif()
+
+option(LIBXMP_DISABLE_IT   "Disable IT support at libXMP-Lite" OFF)
+option(LIBXMP_DISABLE_DEPACKERS     "Disable archive depackers at libXMP" OFF)
+option(LIBXMP_DISABLE_PROWIZARD     "Disable the ProWizard at libXMP" OFF)
+
+TEST_BIG_ENDIAN(WORDS_BIGENDIAN)
+if(WORDS_BIGENDIAN)
+    add_definitions(-DWORDS_BIGENDIAN=1)
+endif()
+
+if(CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
+    add_definitions(-DDEBUG -D_DEBUG -DDEBUG_NOCOLOR)
+endif()
+
+check_include_file(alloca.h HAVE_ALLOCA_H)
+if(HAVE_ALLOCA_H)
+    add_definitions(-DHAVE_ALLOCA_H=1)
+endif()
+
+check_symbol_exists(round "math.h;tgmath.h" HAVE_ROUND_FUNCTION)
+if(HAVE_ROUND_FUNCTION)
+    add_definitions(-DHAVE_ROUND=1)
+endif()
+
+check_symbol_exists(powf "math.h;tgmath.h" HAVE_POWF_FUNCTION)
+if(HAVE_POWF_FUNCTION)
+    add_definitions(-DHAVE_POWF=1)
+endif()
+
+check_symbol_exists(popen "stdio.h" HAVE_POPEN)
+if(HAVE_POPEN)
+    add_definitions(-DHAVE_POPEN=1)
+endif()
+
+check_symbol_exists(fnmatch "fnmatch.h" HAVE_FNMATCH)
+if(HAVE_FNMATCH)
+    add_definitions(-DHAVE_FNMATCH=1)
+endif()
+
+check_symbol_exists(umask "sys/stat.h" HAVE_UMASK)
+if(HAVE_UMASK)
+    add_definitions(-DHAVE_UMASK=1)
+endif()
+
+check_symbol_exists(wait "sys/wait.h" HAVE_WAIT)
+if(HAVE_WAIT)
+    add_definitions(-DHAVE_WAIT=1)
+endif()
+
+check_symbol_exists(pipe "unistd.h" HAVE_PIPE)
+if(HAVE_PIPE)
+    add_definitions(-DHAVE_PIPE=1)
+endif()
+
+check_symbol_exists(fork "unistd.h" HAVE_FORK)
+if(HAVE_FORK)
+    add_definitions(-DHAVE_FORK=1)
+endif()
+
+check_symbol_exists(execvp "unistd.h" HAVE_EXECVP)
+if(HAVE_EXECVP)
+    add_definitions(-DHAVE_EXECVP=1)
+endif()
+
+check_symbol_exists(dup2 "unistd.h" HAVE_DUP2)
+if(HAVE_DUP2)
+    add_definitions(-DHAVE_DUP2=1)
+endif()
+
+check_symbol_exists(mkstemp "stdlib.h" HAVE_MKSTEMP)
+if(HAVE_MKSTEMP)
+    add_definitions(-DHAVE_MKSTEMP=1)
+endif()
+
+check_include_file(dirent.h HAVE_DIRENT)
+if(HAVE_DIRENT)
+    add_definitions(-DHAVE_DIRENT=1)
+endif()
+
+if(UNIX AND NOT APPLE AND NOT HAIKU AND NOT EMSCRIPTEN)
+    find_library(M_LIBRARY m)
+    if(M_LIBRARY) # No need to link it by an absolute path
+        set(M_LIBRARY m)
+    endif()
+    mark_as_advanced(M_LIBRARY)
+else()
+    set(M_LIBRARY "")
+endif()
+
+if("${CMAKE_C_COMPILER_ID}" MATCHES "^(Apple)?Clang$")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-shift-negative-value")
+endif()
+
+if(MSVC)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /wd4244 /wd4018 /wd4996 /wd4048 /wd4267")
+    # set(CMAKE_LINK_LIBRARY_FLAG "${CMAKE_LINK_LIBRARY_FLAG} /wd4273")
+else()
+    if(NOT "${CMAKE_C_COMPILER_ID}" MATCHES "^(Apple)?Clang$")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-but-set-variable -Wno-stringop-truncation")
+    endif()
+    string(REGEX REPLACE "-pedantic" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
```

Comment:
What's with the `-pedantic` joke? You removed it, added it back, and removed it again?

Reply:
Yes, that's a joke, because of the past, and should be removed.